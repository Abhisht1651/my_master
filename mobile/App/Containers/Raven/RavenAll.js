// @flow
import { NavigationContainerProps } from '@react-navigation/native'
import React, { useState, useRef } from 'react'
import { View, Text, TouchableOpacity, Image, FlatList, Platform } from 'react-native'
import { Images } from '../../Themes'
import styles from '../Styles/RavenStyles/RavenAllStyles'
import { BottomNav } from '../../Components/BottomNav'
import RavenFeed from '../../Components/RavenComponents/RavenFeed'
import Dropdown from '../../Components/RavenComponents/Ravendropdown'

type Props = {
  item: {
    images: Array<Image>,
    friendcount: number,
    sent: Boolean,
    sentimages: Array<Image>,
    sentravenperson: string,
    ravenimage: Image
  }
} & NavigationContainerProps
const Data = [
  {
    id: '1',
    name: 'Amelia Jones',
    images: [Images.profile1, Images.profile2, Images.follow1],
    sent: true,
    sentimages: [Images.profile2, Images.follow1, Images.profile1],
    sentravenperson: 'Paul Reith & 10 others',
    friendcount: 69,
    storylineitem: {
      reach: 47,
      heading: 'World',
      storyTitle: 'UK exit from the European Union',
      friendsFollowing: '45 friends Following',
      updatedTime: '2m',
      following: true,
      trending: true,
      storyImage: Images.storyImage
    },
    item: {
      seen: 50,
      total: 500,
      category: 'World',
      storyTitle: 'UK exit from the European Union',
      date: 'Sept 21, 2019',
      description: 'Ray Contreras talks about the different accents',
      source: 'Forbes',
      trending: true,
      storyImage: Images.storyImage
    }
  },
  {
    id: '2',
    name: 'Amelia Jones',
    profilesource: Images.profile1,
    images: [Images.profile1, Images.profile2],
    sent: false,
    friendcount: 69,
    storylineitem: {
      reach: 45,
      heading: 'World',
      storyTitle: 'UK exit from the European Union',
      friendsFollowing: '45 friends Following',
      updatedTime: '2m',
      following: true,
      trending: true,
      storyImage: Images.storyImage
    },
    item: {
      seen: 50,
      total: 500,
      category: 'World',
      storyTitle: 'UK exit from the European Union',
      date: 'Sept 21, 2019',
      description: 'Ray Contreras talks about the different accents',
      source: 'Forbes',
      trending: true,
      storyImage: Images.storyImage
    }
  },
  {
    id: '3',
    name: 'Amelia Jones',
    profilesource: Images.profile1,
    images: [Images.profile1, Images.profile2, Images.follow1],
    sent: false,
    friendcount: 69,
    storylineitem: {
      reach: 45,
      heading: 'World',
      storyTitle: 'UK exit from the European Union',
      friendsFollowing: '45 friends Following',
      updatedTime: '2m',
      following: true,
      trending: true,
      storyImage: Images.storyImage
    },
    item: {
      seen: 50,
      total: 500,
      category: 'World',
      storyTitle: 'UK exit from the European Union',
      date: 'Sept 21, 2019',
      description: 'Ray Contreras talks about the different accents',
      source: 'Forbes',
      trending: true,
      storyImage: Images.storyImage
    }
  },
  {
    id: '4',
    name: 'Amelia Jones',
    profilesource: Images.profile1,
    images: [Images.profile1, Images.profile2, Images.follow1],
    sent: false,
    friendcount: 69,
    storylineitem: {
      reach: 45,
      heading: 'World',
      storyTitle: 'UK exit from the European Union',
      friendsFollowing: '45 friends Following',
      updatedTime: '2m',
      following: true,
      trending: true,
      storyImage: Images.storyImage
    },
    item: {
      seen: 50,
      total: 500,
      category: 'World',
      storyTitle: 'UK exit from the European Union',
      date: 'Sept 21, 2019',
      description: 'Ray Contreras talks about the different accents',
      source: 'Forbes',
      trending: true,
      storyImage: Images.storyImage
    }
  },
  {
    id: '5',
    name: 'Amelia Jones',
    profilesource: Images.profile1,
    images: [Images.profile1, Images.profile2, Images.follow1],
    sent: false,
    friendcount: 69,
    storylineitem: {
      reach: 45,
      heading: 'World',
      storyTitle: 'UK exit from the European Union',
      friendsFollowing: '45 friends Following',
      updatedTime: '2m',
      following: true,
      trending: true,
      storyImage: Images.storyImage
    },
    item: {
      seen: 50,
      total: 500,
      category: 'World',
      storyTitle: 'UK exit from the European Union',
      date: 'Sept 21, 2019',
      description: 'Ray Contreras talks about the different accents',
      source: 'Forbes',
      trending: true,
      storyImage: Images.storyImage
    }
  },
  {
    id: '6',
    name: 'Amelia Jones',
    profilesource: Images.profile1,
    images: [Images.profile1, Images.profile2, Images.follow1],
    sent: false,
    friendcount: 69,
    storylineitem: {
      reach: 45,
      heading: 'World',
      storyTitle: 'UK exit from the European Union',
      friendsFollowing: 45,
      updatedTime: '2m',
      following: true,
      trending: true,
      storyImage: Images.storyImage
    },
    item: {
      seen: 50,
      total: 500,
      category: 'World',
      storyTitle: 'UK exit from the European Union',
      date: 'Sept 21, 2019',
      description: 'Ray Contreras talks about the different accents',
      source: 'Forbes',
      trending: true,
      storyImage: Images.storyImage
    }
  },
  {
    id: '7',
    name: 'Amelia Jones',
    ravenimage: Images.lightfill,
    sentimages: [Images.profile2, Images.profile1],
    sentravenperson: 'Amelia Jones',
    sent: true,
    friendcount: 69,
    storylineitem: {
      reach: 45,
      heading: 'World',
      storyTitle: 'UK exit from the European Union',
      friendsFollowing: 45,
      updatedTime: '2m',
      following: true,
      trending: true,
      storyImage: Images.storyImage
    },
    item: {
      seen: 50,
      total: 500,
      category: 'World',
      storyTitle: 'UK exit from the European Union',
      date: 'Sept 21, 2019',
      description: 'Ray Contreras talks about the different accents',
      source: 'Forbes',
      trending: true,
      storyImage: Images.storyImage
    }
  },
  {
    id: '8',
    name: 'Amelia Jones',
    profilesource: Images.profile1,
    images: [Images.profile1, Images.profile2, Images.follow1],
    sent: false,
    friendcount: 69,
    storylineitem: {
      reach: 45,
      heading: 'World',
      storyTitle: 'UK exit from the European Union',
      friendsFollowing: 45,
      updatedTime: '2m',
      following: true,
      trending: true,
      storyImage: Images.storyImage
    },
    item: {
      seen: 50,
      total: 500,
      category: 'World',
      storyTitle: 'UK exit from the European Union',
      date: 'Sept 21, 2019',
      description: 'Ray Contreras talks about the different accents',
      source: 'Forbes',
      trending: true,
      storyImage: Images.storyImage
    }
  },
  {
    id: '9',
    name: 'Amelia Jones',
    profilesource: Images.profile1,
    images: [Images.profile1, Images.profile2, Images.follow1],
    sent: false,
    friendcount: 69,
    storylineitem: {
      reach: 45,
      heading: 'World',
      storyTitle: 'UK exit from the European Union',
      friendsFollowing: '45 friends Following',
      updatedTime: '2m',
      following: true,
      trending: true,
      storyImage: Images.storyImage
    },
    item: {
      seen: 50,
      total: 500,
      category: 'World',
      storyTitle: 'UK exit from the European Union',
      date: 'Sept 21, 2019',
      description: 'Ray Contreras talks about the different accents',
      source: 'Forbes',
      trending: true,
      storyImage: Images.storyImage
    }
  }
]

const RavenAll = (props: Props) => {
  const { navigation } = props
  const [modal, setmodal] = useState(false)
  const [newraven] = useState(13)
  const [newmessage] = useState(45)
  const flatlist = useRef<FlatList>()
  const [type, settype] = useState('Filter by')
  const applyModal = () => {
    setmodal(!modal)
  }
  const applyType = (filter) => {
    settype(filter)
    applyModal()
    flatlist.current.scrollToOffset({
      y: 0,
      animated: true
    })
  }
  const renderfeed = ({ item }: Props) => {
    if (
      type === 'All' ||
      type === 'Filter by' ||
      (type === 'Sent' && item.sent === true) ||
      (type === 'Received' && item.sent === false)
    ) {
      return (
        <View>
          <RavenFeed
            images={item.images}
            storylineitem={item.storylineitem}
            friendcount={item.friendcount}
            sent={item.sent}
            sentimages={item.sentimages}
            ravensentperson={item.sentravenperson}
            ravenfriends={() => props.navigation.navigate('ListOfFriends')}
            sendmorefriends={() => props.navigation.navigate('SendRaven', { item: item.item })}
            ravenimage={item.ravenimage}
          />
        </View>
      )
    }
    return <View />
  }
  return (
    <View style={styles.container}>
      <View style={styles.supercontainer}>
        {Platform.OS === 'ios' && <View style={styles.topcover} />}
        <View style={styles.container1}>
          <View style={styles.ravencontext}>
            <TouchableOpacity>
              <View>
                <View style={styles.ravencont}>
                  <Text style={styles.raventext}>Ravens</Text>
                </View>
                {newraven > 0 && (
                  <View style={styles.newraven}>
                    <Text style={styles.newravencount}>{newraven}</Text>
                  </View>
                )}
              </View>
            </TouchableOpacity>
          </View>
          <View style={styles.messagecontext}>
            <TouchableOpacity onPress={() => navigation.navigate('Message')}>
              <View style={styles.messagecont}>
                <Text style={styles.messagetext}>Messages</Text>
              </View>
              {newmessage > 0 && (
                <View style={styles.newmessage}>
                  <Text style={styles.newmessagecount}>{newmessage}</Text>
                </View>
              )}
            </TouchableOpacity>
          </View>
          <View style={styles.writecont}>
            <TouchableOpacity onPress={() => props.navigation.navigate('SendNewRaven')}>
              <Image source={Images.write} style={styles.writeimage} />
            </TouchableOpacity>
          </View>
        </View>
        <View style={styles.container2} />
        <View style={styles.container3}>
          <FlatList
            ref={flatlist}
            data={Data}
            renderItem={renderfeed}
            keyExtractor={(item) => item.id}
            showsVerticalScrollIndicator={false}
          />
        </View>
        <View style={Platform.OS === 'ios' ? styles.iconsvg2 : styles.iconsvg}>
          <Dropdown applyModal={applyModal} applyType={applyType} type={type} modal={modal} />
        </View>
      </View>
      <View style={styles.footer}>
        <BottomNav
          chats
          activityonPress={() => props.navigation.navigate('ActivityFeed')}
          newsfeedonPress={() => props.navigation.navigate('NewsFeed')}
          searchonPress={() => props.navigation.navigate('Discover')}
        />
      </View>
    </View>
  )
}

export default RavenAll
