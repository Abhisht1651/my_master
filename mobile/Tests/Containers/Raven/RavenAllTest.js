import React from 'react'
import { shallow } from 'enzyme'
import { TouchableOpacity, Text, FlatList } from 'react-native'
import RavenAll from '../../../App/Containers/Raven/RavenAll'
// Test for RavenAll (snapshot and props)
describe('RavenAll', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallow(<RavenAll />) // no compile-time error
  })
  it('RavenAll component renders correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
  it('RavenAll BottomNav', () => {
    expect(wrapper.find('BottomNav')).toHaveLength(1)
  })
  it('should render 3 TouchableOpacity', () => {
    expect(wrapper.find(TouchableOpacity)).toHaveLength(3)
  })
  it('should render 4 Text', () => {
    expect(wrapper.find(Text)).toHaveLength(4)
  })
  it('should render 1 FlatList', () => {
    expect(wrapper.find(FlatList)).toHaveLength(1)
  })
})
