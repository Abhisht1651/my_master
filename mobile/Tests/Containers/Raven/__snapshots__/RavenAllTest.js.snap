// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RavenAll RavenAll component renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 693,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "white",
          "height": 58.623046875,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderBottomColor": "#E6ECED",
          "borderBottomWidth": 1.7369791666666665,
          "height": 112.90364583333334,
        }
      }
    >
      <View
        style={
          Object {
            "marginLeft": 50,
            "marginTop": 22,
            "width": 164,
          }
        }
      >
        <ForwardRef>
          <View>
            <View
              style={
                Object {
                  "borderBottomColor": "#7B46E4",
                  "borderBottomWidth": 3.39440203562341,
                  "borderRadius": 1.697201017811705,
                  "height": 60,
                  "width": 107.99999999999999,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#22226B",
                    "fontFamily": "BananaGrotesk-Medium",
                    "fontSize": 32,
                    "letterSpacing": -0.107421875,
                    "lineHeight": 53.99999999999999,
                    "textAlign": "center",
                  }
                }
              >
                Ravens
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#46C3F4",
                  "borderRadius": 16,
                  "flex": 1,
                  "height": 36,
                  "justifyContent": "center",
                  "left": 116,
                  "position": "absolute",
                  "top": 10,
                  "width": 56.00000000000001,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#FFFFFF",
                    "fontFamily": "BananaGrotesk-Regular",
                    "fontSize": 24,
                    "letterSpacing": -0.14,
                    "lineHeight": 28.000000000000004,
                  }
                }
              >
                13
              </Text>
            </View>
          </View>
        </ForwardRef>
      </View>
      <View
        style={
          Object {
            "marginLeft": 295,
            "marginTop": 22,
            "position": "absolute",
          }
        }
      >
        <ForwardRef
          onPress={[Function]}
        >
          <View
            style={
              Object {
                "width": 160,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#A4A4C0",
                  "fontFamily": "BananaGrotesk-Medium",
                  "fontSize": 32,
                  "letterSpacing": -0.22,
                  "lineHeight": 53.99999999999999,
                  "textAlign": "center",
                }
              }
            >
              Messages
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#46C3F4",
                "borderRadius": 16,
                "flex": 1,
                "height": 36,
                "justifyContent": "center",
                "left": 164,
                "position": "absolute",
                "top": 8.684895833333334,
                "width": 56.00000000000001,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#FFFFFF",
                  "fontFamily": "BananaGrotesk-Regular",
                  "fontSize": 24,
                  "letterSpacing": -0.14,
                  "lineHeight": 28.000000000000004,
                }
              }
            >
              45
            </Text>
          </View>
        </ForwardRef>
      </View>
      <View
        style={
          Object {
            "marginTop": 22,
            "position": "absolute",
            "right": 50,
            "width": 60,
          }
        }
      >
        <ForwardRef
          onPress={[Function]}
        >
          <Image
            source={
              Object {
                "default": "",
              }
            }
            style={
              Object {
                "height": 48.63541666666667,
                "width": 52.109375,
              }
            }
          />
        </ForwardRef>
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderBottomColor": "#E6ECED",
          "borderBottomWidth": 1.7369791666666665,
          "height": 69.47916666666667,
          "justifyContent": "center",
          "zIndex": 2,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#F7F7F7",
          "flex": 1,
        }
      }
    >
      <FlatList
        data={
          Array [
            Object {
              "friendcount": 69,
              "id": "1",
              "images": Array [
                Object {
                  "default": "",
                },
                Object {
                  "default": "",
                },
                Object {
                  "default": "",
                },
              ],
              "item": Object {
                "category": "World",
                "date": "Sept 21, 2019",
                "description": "Ray Contreras talks about the different accents",
                "seen": 50,
                "source": "Forbes",
                "storyImage": Object {
                  "default": "",
                },
                "storyTitle": "UK exit from the European Union",
                "total": 500,
                "trending": true,
              },
              "name": "Amelia Jones",
              "sent": true,
              "sentimages": Array [
                Object {
                  "default": "",
                },
                Object {
                  "default": "",
                },
                Object {
                  "default": "",
                },
              ],
              "sentravenperson": "Paul Reith & 10 others",
              "storylineitem": Object {
                "following": true,
                "friendsFollowing": "45 friends Following",
                "heading": "World",
                "reach": 47,
                "storyImage": Object {
                  "default": "",
                },
                "storyTitle": "UK exit from the European Union",
                "trending": true,
                "updatedTime": "2m",
              },
            },
            Object {
              "friendcount": 69,
              "id": "2",
              "images": Array [
                Object {
                  "default": "",
                },
                Object {
                  "default": "",
                },
              ],
              "item": Object {
                "category": "World",
                "date": "Sept 21, 2019",
                "description": "Ray Contreras talks about the different accents",
                "seen": 50,
                "source": "Forbes",
                "storyImage": Object {
                  "default": "",
                },
                "storyTitle": "UK exit from the European Union",
                "total": 500,
                "trending": true,
              },
              "name": "Amelia Jones",
              "profilesource": Object {
                "default": "",
              },
              "sent": false,
              "storylineitem": Object {
                "following": true,
                "friendsFollowing": "45 friends Following",
                "heading": "World",
                "reach": 45,
                "storyImage": Object {
                  "default": "",
                },
                "storyTitle": "UK exit from the European Union",
                "trending": true,
                "updatedTime": "2m",
              },
            },
            Object {
              "friendcount": 69,
              "id": "3",
              "images": Array [
                Object {
                  "default": "",
                },
                Object {
                  "default": "",
                },
                Object {
                  "default": "",
                },
              ],
              "item": Object {
                "category": "World",
                "date": "Sept 21, 2019",
                "description": "Ray Contreras talks about the different accents",
                "seen": 50,
                "source": "Forbes",
                "storyImage": Object {
                  "default": "",
                },
                "storyTitle": "UK exit from the European Union",
                "total": 500,
                "trending": true,
              },
              "name": "Amelia Jones",
              "profilesource": Object {
                "default": "",
              },
              "sent": false,
              "storylineitem": Object {
                "following": true,
                "friendsFollowing": "45 friends Following",
                "heading": "World",
                "reach": 45,
                "storyImage": Object {
                  "default": "",
                },
                "storyTitle": "UK exit from the European Union",
                "trending": true,
                "updatedTime": "2m",
              },
            },
            Object {
              "friendcount": 69,
              "id": "4",
              "images": Array [
                Object {
                  "default": "",
                },
                Object {
                  "default": "",
                },
                Object {
                  "default": "",
                },
              ],
              "item": Object {
                "category": "World",
                "date": "Sept 21, 2019",
                "description": "Ray Contreras talks about the different accents",
                "seen": 50,
                "source": "Forbes",
                "storyImage": Object {
                  "default": "",
                },
                "storyTitle": "UK exit from the European Union",
                "total": 500,
                "trending": true,
              },
              "name": "Amelia Jones",
              "profilesource": Object {
                "default": "",
              },
              "sent": false,
              "storylineitem": Object {
                "following": true,
                "friendsFollowing": "45 friends Following",
                "heading": "World",
                "reach": 45,
                "storyImage": Object {
                  "default": "",
                },
                "storyTitle": "UK exit from the European Union",
                "trending": true,
                "updatedTime": "2m",
              },
            },
            Object {
              "friendcount": 69,
              "id": "5",
              "images": Array [
                Object {
                  "default": "",
                },
                Object {
                  "default": "",
                },
                Object {
                  "default": "",
                },
              ],
              "item": Object {
                "category": "World",
                "date": "Sept 21, 2019",
                "description": "Ray Contreras talks about the different accents",
                "seen": 50,
                "source": "Forbes",
                "storyImage": Object {
                  "default": "",
                },
                "storyTitle": "UK exit from the European Union",
                "total": 500,
                "trending": true,
              },
              "name": "Amelia Jones",
              "profilesource": Object {
                "default": "",
              },
              "sent": false,
              "storylineitem": Object {
                "following": true,
                "friendsFollowing": "45 friends Following",
                "heading": "World",
                "reach": 45,
                "storyImage": Object {
                  "default": "",
                },
                "storyTitle": "UK exit from the European Union",
                "trending": true,
                "updatedTime": "2m",
              },
            },
            Object {
              "friendcount": 69,
              "id": "6",
              "images": Array [
                Object {
                  "default": "",
                },
                Object {
                  "default": "",
                },
                Object {
                  "default": "",
                },
              ],
              "item": Object {
                "category": "World",
                "date": "Sept 21, 2019",
                "description": "Ray Contreras talks about the different accents",
                "seen": 50,
                "source": "Forbes",
                "storyImage": Object {
                  "default": "",
                },
                "storyTitle": "UK exit from the European Union",
                "total": 500,
                "trending": true,
              },
              "name": "Amelia Jones",
              "profilesource": Object {
                "default": "",
              },
              "sent": false,
              "storylineitem": Object {
                "following": true,
                "friendsFollowing": 45,
                "heading": "World",
                "reach": 45,
                "storyImage": Object {
                  "default": "",
                },
                "storyTitle": "UK exit from the European Union",
                "trending": true,
                "updatedTime": "2m",
              },
            },
            Object {
              "friendcount": 69,
              "id": "7",
              "item": Object {
                "category": "World",
                "date": "Sept 21, 2019",
                "description": "Ray Contreras talks about the different accents",
                "seen": 50,
                "source": "Forbes",
                "storyImage": Object {
                  "default": "",
                },
                "storyTitle": "UK exit from the European Union",
                "total": 500,
                "trending": true,
              },
              "name": "Amelia Jones",
              "ravenimage": Object {
                "default": "",
              },
              "sent": true,
              "sentimages": Array [
                Object {
                  "default": "",
                },
                Object {
                  "default": "",
                },
              ],
              "sentravenperson": "Amelia Jones",
              "storylineitem": Object {
                "following": true,
                "friendsFollowing": 45,
                "heading": "World",
                "reach": 45,
                "storyImage": Object {
                  "default": "",
                },
                "storyTitle": "UK exit from the European Union",
                "trending": true,
                "updatedTime": "2m",
              },
            },
            Object {
              "friendcount": 69,
              "id": "8",
              "images": Array [
                Object {
                  "default": "",
                },
                Object {
                  "default": "",
                },
                Object {
                  "default": "",
                },
              ],
              "item": Object {
                "category": "World",
                "date": "Sept 21, 2019",
                "description": "Ray Contreras talks about the different accents",
                "seen": 50,
                "source": "Forbes",
                "storyImage": Object {
                  "default": "",
                },
                "storyTitle": "UK exit from the European Union",
                "total": 500,
                "trending": true,
              },
              "name": "Amelia Jones",
              "profilesource": Object {
                "default": "",
              },
              "sent": false,
              "storylineitem": Object {
                "following": true,
                "friendsFollowing": 45,
                "heading": "World",
                "reach": 45,
                "storyImage": Object {
                  "default": "",
                },
                "storyTitle": "UK exit from the European Union",
                "trending": true,
                "updatedTime": "2m",
              },
            },
            Object {
              "friendcount": 69,
              "id": "9",
              "images": Array [
                Object {
                  "default": "",
                },
                Object {
                  "default": "",
                },
                Object {
                  "default": "",
                },
              ],
              "item": Object {
                "category": "World",
                "date": "Sept 21, 2019",
                "description": "Ray Contreras talks about the different accents",
                "seen": 50,
                "source": "Forbes",
                "storyImage": Object {
                  "default": "",
                },
                "storyTitle": "UK exit from the European Union",
                "total": 500,
                "trending": true,
              },
              "name": "Amelia Jones",
              "profilesource": Object {
                "default": "",
              },
              "sent": false,
              "storylineitem": Object {
                "following": true,
                "friendsFollowing": "45 friends Following",
                "heading": "World",
                "reach": 45,
                "storyImage": Object {
                  "default": "",
                },
                "storyTitle": "UK exit from the European Union",
                "trending": true,
                "updatedTime": "2m",
              },
            },
          ]
        }
        disableVirtualization={false}
        horizontal={false}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        numColumns={1}
        onEndReachedThreshold={2}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        showsVerticalScrollIndicator={false}
        updateCellsBatchingPeriod={50}
        windowSize={21}
      />
    </View>
    <View
      style={
        Object {
          "position": "absolute",
          "right": 30,
          "top": 193.67317708333331,
          "zIndex": 3,
        }
      }
    >
      <DropDown2
        applyModal={[Function]}
        applyType={[Function]}
        modal={false}
        type="Filter by"
      />
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 75,
      }
    }
  >
    <BottomNav
      activity={false}
      activityonPress={[Function]}
      chats={true}
      chatsonPress={[Function]}
      newsfeed={false}
      newsfeedonPress={[Function]}
      profile={false}
      profileonPress={[Function]}
      search={false}
      searchonPress={[Function]}
    />
  </View>
</View>
`;
